ee/server/service/mdm.go

```diff
@@ -1055,6 +1055,63 @@ func (svc *Service) GetMDMDiskEncryptionSummary(ctx context.Context, teamID *uin
 	}, nil
 }
 
+func (svc *Service) mdmAppleEditedMacOSUpdates(ctx context.Context, teamID *uint, updates fleet.MacOSUpdates) error {
+	// TODO: is there a notion of "DDM enabled" or not, where the DDM profile
+	// should not be created?
+
+	if updates.MinimumVersion.Value == "" {
+		// OS updates disabled, remove the profile
+		if err := svc.ds.DeleteMDMAppleDeclarationByName(ctx, teamID, mdm.FleetMacOSUpdatesProfileName); err != nil {
+			return err
+		}
+		var globalOrTeamID uint
+		if teamID != nil {
+			globalOrTeamID = *teamID
+		}
+		if err := svc.ds.BulkSetPendingMDMHostProfiles(ctx, nil, []uint{globalOrTeamID}, nil, nil); err != nil {
+			return ctxerr.Wrap(ctx, err, "bulk set pending host profiles")
+		}
+		return nil
+	}
+
+	// OS updates enabled, create or update the profile with the current settings
+
+	const macOSSoftwareUpdateType = `com.apple.configuration.softwareupdate.enforcement.specific`
+	ident := uuid.NewString()
+	// TODO(mna): is that correct payload? Identifier is a uuid? Is it ok if it
+	// changes on every update?
+	rawDecl := []byte(fmt.Sprintf(`{
+	"Identifier": %q,
+	"Type": %q,
+	"Payload": {
+		"TargetOSVersion": %q,
+		"TargetLocalDateTime ": "2024-03-01T12:00:00,"
+	}
+}`, ident, macOSSoftwareUpdateType, updates.MinimumVersion.Value))
+	d := fleet.NewMDMAppleDeclaration(rawDecl, teamID, mdm.FleetMacOSUpdatesProfileName, macOSSoftwareUpdateType, ident)
+
+	// associate the profile with the built-in label that ensures the host is on
+	// macOS 14+ to receive that profile
+	lblIDs, err := svc.ds.LabelIDsByName(ctx, []string{fleet.BuiltinLabelMacOS14Plus})
+	if err != nil {
+		return err
+	}
+	d.Labels = []fleet.ConfigurationProfileLabel{
+		{LabelName: fleet.BuiltinLabelMacOS14Plus, LabelID: lblIDs[fleet.BuiltinLabelMacOS14Plus]},
+	}
+
+	decl, err := svc.ds.SetOrUpdateMDMAppleDeclaration(ctx, d)
+	if err != nil {
+		return err
+	}
+
+	// mark all hosts affected by that profile as pending
+	if err := svc.ds.BulkSetPendingMDMHostProfiles(ctx, nil, nil, []string{decl.DeclarationUUID}, nil); err != nil {
```

Comment:
yes! this does the same as it does for profiles: sets the declaration as "pending" in the UI, but with the right DB conditions so it's still picked in the next cron run.

the only difference: is less efficient for declarations because it'll set _all_ newly updated declarations as pending vs the profile bits where we pick a subset of declarations based on the team/host parameter.

Reply:
Sounds good, thanks! Ideally though, it wouldn't consider the declaration UUID a Windows profile, but I don't think it matters much because it won't match any profile (so, won't end up matching any host) because the prefix letter is different for declarations.

Reply:
very good point, this was made like this in the heat of the moment, but we should improve it.

Reply:
I'll give it a shot as part of this PR.